src/main/java/uk/gov/gchq/palisade/service/request/Policy.java

```diff
@@ -0,0 +1,229 @@
+        requireNonNull(message, "The message cannot be null and should indicate what the rule is doing.");
+        requireNonNull(rule, "Cannot set a null rule.");
+        Rules<T> recordLevelRules = getRecordRules();
+        recordLevelRules.simpleFunctionRule(generateUUID(), rule);
+        addMessage(message, recordLevelRules);
+        return this;
+    }
+
+    public Policy<T> resourceLevelRule(final String message, final Rule<Resource> rule) {
+        requireNonNull(message, "The message cannot be null and should indicate what the rule is doing.");
+        requireNonNull(rule, "Cannot set a null rule.");
+        Rules<Resource> resourceLevelRules = getResourceRules();
+        resourceLevelRules.rule(generateUUID(), rule);
+        addMessage(message, resourceLevelRules);
+        return this;
+    }
+
+    public Policy<T> resourceLevelPredicateRule(final String message, final PredicateRule<Resource> rule) {
+        requireNonNull(message, "The message cannot be null and should indicate what the rule is doing.");
+        requireNonNull(rule, "Cannot set a null rule.");
+        Rules<Resource> resourceLevelRules = getResourceRules();
+        resourceLevelRules.rule(generateUUID(), rule);
+        addMessage(message, resourceLevelRules);
+        return this;
+    }
+
+    public Policy<T> resourceLevelSimplePredicateRule(final String message, final Predicate<Resource> rule) {
+        requireNonNull(message, "The message cannot be null and should indicate what the rule is doing.");
+        requireNonNull(rule, "Cannot set a null rule.");
+        Rules<Resource> resourceLevelRules = getResourceRules();
+        resourceLevelRules.simplePredicateRule(generateUUID(), rule);
+        addMessage(message, resourceLevelRules);
+        return this;
+    }
+
+    public Policy<T> resourceLevelSimpleFunctionRule(final String message, final UnaryOperator<Resource> rule) {
+        requireNonNull(message, "The message cannot be null and should indicate what the rule is doing.");
+        requireNonNull(rule, "Cannot set a null rule.");
+        Rules<Resource> resourceLevelRules = getResourceRules();
+        resourceLevelRules.simpleFunctionRule(generateUUID(), rule);
+        addMessage(message, resourceLevelRules);
+        return this;
+    }
+
+    public User getOwner() {
+        requireNonNull(owner, "The owner has not been set.");
+        return owner;
+    }
+
+    public void setOwner(final User owner) {
+        owner(owner);
+    }
+
+    public Policy<T> owner(final User owner) {
+        requireNonNull(owner, "The owner cannot be set to null.");
+        this.owner = owner;
+        return this;
+    }
+
+    @JsonGetter("owner")
+    User getNullableOwner() {
+        return owner;
+    }
+
+    @Override
+    @Generated
+    public boolean equals(final Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
```

Comment:
Could you update the code-generation xmls? This should be an instanceOf thing rather than a getClass thing